drivers/sensor/mhz19b/mhz19b.c

```diff
@@ -0,0 +1,324 @@
+		return -ENOTSUP;
+	}
+
+	if (!data->data_valid) {
+		LOG_DBG("Checksum mismatch");
+		return -EINVAL;
+	}
+
+	val->val1 = (int32_t)data->data;
+	val->val2 = 0;
+
+	return 0;
+}
+
+static int mhz19b_attr_set(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, const struct sensor_value *val)
+{
+	if (chan != SENSOR_CHAN_CO2) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		if (val->val1 == 2000) {
+			LOG_DBG("Configured range to %d", val->val1);
+			return mhz19b_send_config(dev, MHZ19B_CMD_SET_RANGE_2000);
+		} else if (val->val1 == 5000) {
+			LOG_DBG("Configured range to %d", val->val1);
+			return mhz19b_send_config(dev, MHZ19B_CMD_SET_RANGE_5000);
+		} else if (val->val1 == 10000) {
+			LOG_DBG("Configured range to %d", val->val1);
+			return mhz19b_send_config(dev, MHZ19B_CMD_SET_RANGE_10000);
+		} else {
+			return -ENOTSUP;
+		}
+
+	case SENSOR_ATTR_MHZ19B_ABC:
+		if (val->val1 == 0) {
+			LOG_DBG("%s ABC", "Disabled");
+			return mhz19b_send_config(dev, MHZ19B_CMD_SET_ABC_OFF);
+		} else if (val->val1 == 1) {
+			LOG_DBG("%s ABC", "Enabled");
+			return mhz19b_send_config(dev, MHZ19B_CMD_SET_ABC_ON);
+		} else {
+			return -ENOTSUP;
+		}
+
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int mhz19b_attr_get(const struct device *dev, enum sensor_channel chan,
+			   enum sensor_attribute attr, struct sensor_value *val)
+{
+	struct mhz19b_data *data = dev->data;
+	int ret;
+
+	if (chan != SENSOR_CHAN_CO2) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_FULL_SCALE:
+		ret = mhz19b_poll_data(dev, MHZ19B_CMD_GET_RANGE);
+		val->val1 = (int32_t)data->data;
+		val->val2 = 0;
+		return ret;
+	case SENSOR_ATTR_MHZ19B_ABC:
+		ret = mhz19b_poll_data(dev, MHZ19B_CMD_GET_ABC);
+		val->val1 = (int32_t)data->data;
+		val->val2 = 0;
+		return ret;
+	default:
+		return -ENOTSUP;
+	}
+}
+
+static int mhz19b_trigger_set(const struct device *dev, const struct sensor_trigger *trig,
+			      sensor_trigger_handler_t handler)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(trig);
+	ARG_UNUSED(handler);
+
+	return -ENOTSUP;
+}
+
+static int mhz19b_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_VOLTAGE && chan != SENSOR_CHAN_POWER &&
+	    chan != SENSOR_CHAN_CURRENT) {
+		return -ENOTSUP;
+	}
+
+	return mhz19b_poll_data(dev, MHZ19B_CMD_GET_CO2);
+}
+
+static const struct sensor_driver_api mhz19b_api_funcs = {
+	.attr_set = mhz19b_attr_set,
+	.attr_get = mhz19b_attr_get,
+	.trigger_set = mhz19b_trigger_set,
+	.sample_fetch = mhz19b_sample_fetch,
+	.channel_get = mhz19b_channel_get,
+};
+
+static void mhz19b_uart_isr(const struct device *uart_dev, void *user_data)
+{
+	const struct device *dev = user_data;
+	struct mhz19b_data *data = dev->data;
+
+	ARG_UNUSED(user_data);
+
+	if (uart_dev == NULL) {
+		return;
+	}
+
+	if (!uart_irq_update(uart_dev)) {
+		return;
+	}
+
+	if (uart_irq_rx_ready(uart_dev)) {
```

Comment:
Maybe declare `ret` variable and do:

<pre>
ret = uart_irq_rx_ready(uart_dev);
if (ret < 0) {
     return ret;
}
</pre>

Reply:
> Maybe declare ret variable and do

This function has `void` return type to match the `uart_irq_callback_user_data_t` signature; there's nothing to return.

Reply:
Right, missed that.